### PR TITLE
syslog-ng: update to 3.17.2

### PR DESCRIPTION
**Pull request by MikePetullo**
Signed-off-by: W. Michael Petullo <mike@flyn.org>

Maintainer: me
Compile tested: x86_64, master f4d30476

Description:

### CODE REVIEW

**Comment by pprindeville** [comment]
Do we know why the build is failing?

**Comment by cshoredaniel** [comment]
Wonder if it's something to do with util-linux upgrade a day or two ago in core?  (Either commit needed or commit breaks)

**Comment by pprindeville** [comment]
Installed this, but I'm seeing:

```
Aug 19 15:40:28 OpenWrt syslog-ng[5533]: syslog-ng shutting down; version='3.16.1'
Aug 19 15:40:29 OpenWrt syslog-ng[13545]: syslog-ng starting up; version='3.17.2'
Aug 19 15:42:28 OpenWrt syslog-ng[13545]: Error opening plugin module; module='tfgetent', error='Error relocating /usr/lib/syslog-ng/libtfgetent.so: getprotobynumber_r: symbol not found'
Aug 19 15:42:28 OpenWrt syslog-ng[13545]: Configuration reload request received, reloading configuration;
Aug 19 15:42:28 OpenWrt syslog-ng[13545]: Configuration reload finished;
```

That seems to be part of `glibc` but not `musl`.

**Comment by pprindeville** [comment]
See https://github.com/balabit/syslog-ng/issues/2238

**Comment by pprindeville** [comment]
> what did you do to cause syslog-ng to load the tfgetent module? 

Actually nothing special.  I

1. rebuilt the package after rebasing;
2. copied over the `.ipk` to a production router;
3. did a `opkg install syslog-ng_3.17.2-1_x86_64.pkg`;
4. did `/etc/init.d/syslog-ng reload`;

Ah... if I do a `restart` instead I don't get this issue.  So it looks like the old process was still running but tried to dyn-link to the new libraries...

**Comment by pprindeville** [comment]
@MikePetullo Should the `postinst` script (or one of the others) cause the service to restart on update?

**Comment by MikePetullo** [comment]
Added patch from https://github.com/balabit/syslog-ng/pull/2244/ after testing on x86_64.

**Comment by MikePetullo** [comment]
Updated configuration to reference 3.17.

File: admin/syslog-ng/files/syslog-ng.conf [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@version:3.9
+@version: 3.17
```
Comment (pprindeville) [inline admin/syslog-ng/files/syslog-ng.conf]: Please remove the space.

**Comment by MikePetullo** [comment]
I put the space in there on purpose. I think the space makes the statement more readable. A similar space is present in several of the examples provided by the syslog-ng folks. For example, see:

https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.17/administration-guide/12

**Comment by pprindeville** [comment]
Not sure this is the right fix, but see PR #6869.

**Comment by MikePetullo** [comment]
@thess, can we merge this now? Travis now succeeds thanks to @pprindeville's #6869.

**Comment by dibdot** [comment]
Thanks!

**Pull request merged**